app/Scripts/RockTurtle.gd

```diff
@@ -0,0 +1,49 @@
+extends Node2D
+
+
+var Target: Area2D
+var TargetDir: Vector2
+var HasTarget := false
+onready var attackAnimationPlayer = $AnimationPlayer
+onready var attackTimer = $AttackTimer
+onready var _sprite = $Sprite
+func _ready():
+	_sprite = "res://Assets/01.png"
+func _process(delta):
+	if Target != null:
+		TargetDir = self.position - Target.position
+		
+func _on_TurtleVision_area_entered(area):
+	if HasTarget == false:
+		Target = area
+		HasTarget = true
+		
+		attackTimer.start(0)
+		print(TargetDir, area)
+		
+
+
+func _on_TurtleVision_area_exited(area):
+	if area == Target:
+		print("exit")
+		if HasTarget == true:
```

Comment:
we don't need a separate HasTarget variable, you can just check if `is_instance_valid(Target)` I believe